csharp/lib/ConnectionConfiguration.cs

```diff
@@ -0,0 +1,563 @@
+        // TODO possible options : list and array
+        #endregion
+        #region TLS
+        /// <summary>
+        /// Configure whether communication with the server should use Transport Level Security.<br />
+        /// Should match the TLS configuration of the server/cluster, otherwise the connection attempt will fail.
+        /// </summary>
+        public TlsMode TlsMode
+        {
+            set
+            {
+                Config.TlsMode = value;
+            }
+        }
+        /// <inheritdoc cref="TlsMode"/>
+        public T WithTlsMode(TlsMode tls_mode)
+        {
+            TlsMode = tls_mode;
+            return (T)this;
+        }
+        /// <inheritdoc cref="TlsMode"/>
+        public T With(TlsMode tls_mode)
+        {
+            return WithTlsMode(tls_mode);
+        }
+        #endregion
+        #region Request Timeout
+        /// <summary>
+        /// The duration in milliseconds that the client should wait for a request to complete. This
+        /// duration encompasses sending the request, awaiting for a response from the server, and any
+        /// required reconnections or retries. If the specified timeout is exceeded for a pending request,
+        /// it will result in a timeout error. If not set, a default value will be used.
+        /// </summary>
+        public uint RequestTimeout
+        {
+            set
+            {
+                Config.RequestTimeout = value;
+            }
+        }
+        /// <inheritdoc cref="RequestTimeout"/>
+        public T WithRequestTimeout(uint request_timeout)
+        {
+            RequestTimeout = request_timeout;
+            return (T)this;
+        }
+        #endregion
+        #region Read From
+        /// <summary>
+        /// Configure the client's read from strategy. If not set, <seealso cref="ReadFrom.Primary"/> will be used.
+        /// </summary>
+        public ReadFrom ReadFrom
+        {
+            set
+            {
+                Config.ReadFrom = value;
+            }
+        }
+        /// <inheritdoc cref="ReadFrom"/>
+        public T WithReadFrom(ReadFrom read_from)
+        {
+            ReadFrom = read_from;
+            return (T)this;
+        }
+        /// <inheritdoc cref="ReadFrom"/>
+        public T With(ReadFrom read_from)
+        {
+            return WithReadFrom(read_from);
+        }
+        #endregion
+        #region Authentication
+        /// <summary>
+        /// Configure credentials for authentication process. If none are set, the client will not authenticate itself with the server.
+        /// </summary>
+        /// <value>
+        /// <c>username</c> The username that will be used for authenticating connections to the Redis servers. If not supplied, <c>"default"</c> will be used.<br />
+        /// <c>password</c> The password that will be used for authenticating connections to the Redis servers.
+        /// </value>
+        public (string? username, string? password) Authentication
```

Comment:
I'm not familiar with C# but it looks like ? indicates the variable is nullable. Isn't password a required field? Its marked with @NonNull in Java, so should we make password NonNull here?

Reply:
Good point thanks

Reply:
Thanks, fixed in eee257f1